src/OpenTelemetry.ResourceDetectors.Container/KubeApiConnector.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+#if !NETFRAMEWORK
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.ResourceDetectors.Container.Http;
+
+namespace OpenTelemetry.ResourceDetectors.Container;
+
+internal class KubeApiConnector
+{
+    protected const int MaxAttempts = 3;
+
+    protected static readonly TimeSpan FiveSeconds = TimeSpan.FromSeconds(5);
+
+    protected static readonly HashSet<HttpStatusCode> AcceptableResponseCodes =
+        [HttpStatusCode.OK, HttpStatusCode.Unauthorized, HttpStatusCode.Forbidden];
+
+    protected static HttpClient httpClient = new();
+
+    protected int connectionTimeout = 5;
+
+    public KubeApiConnector(string? kubeHost, string? kubePort, string? certFile, string? token, string? nameSpace, string? kubeHostName)
+    {
+        this.Target = new Uri($"https://{kubeHost}:{kubePort}/api/v1/namespaces/{nameSpace}/pods/{kubeHostName}");
+
+        if (certFile == null)
+        {
+            return;
+        }
+
+        using HttpClientHandler? clientHandler = Handler.Create(certFile);
+        if (clientHandler == null)
+        {
+            return;
+        }
+
+        clientHandler.CheckCertificateRevocationList = true;
+
+        httpClient = new HttpClient(clientHandler);
+        httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+        httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+    }
+
+    public Uri Target { get; }
+
+    public string ExecuteRequest()
+    {
+        // Give arbitrary amount of time for the kube api to update container status
+        Thread.Sleep(TimeSpan.FromSeconds(FiveSeconds.TotalSeconds));
```

Comment:
Do we really need to suspend the thread here (especially 5 seconds which seems really long). Considering that the container will already be running here because this code will be executed at runtime, I imagine the value will already be updated at K8s level.